Terms/SolrRegexTermsService.php

```diff
@@ -24,10 +24,9 @@ public function fetchTerms(SimpleQueryInterface $query)
             return new EmptyTermsResult();
         }
 
-        $suggestQuery
-            ->setFields($field)
-            ->setRegex($query->getSearchTerm())
-            ->setSort('count');
+        $suggestQuery->setFields($field);
+        $suggestQuery->setRegex($this->escapeTerm($query->getSearchTerm()));
```

Comment:
This field expects a regular expression value, *not* a Lucene expression.

https://cwiki.apache.org/confluence/display/solr/The+Terms+Component (under terms.regex)

To escape, try using `preg_quote()` or similar.